connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -236,6 +238,22 @@ public Converter newConverter(AbstractConfig config, String classPropertyName, C
         Map<String, Object> converterConfig = config.originalsWithPrefix(configPrefix);
         log.debug("Configuring the {} converter with configuration:{}{}",
                   isKeyConverter ? "key" : "value", System.lineSeparator(), converterConfig);
+
+        // Have to override schemas.enable from true to false for internal JSON converters
+        // Don't have to warn the user about anything since all deprecation warnings take place in the
+        // WorkerConfig class
+        if (plugin instanceof JsonConverter) {
```

Comment:
Is this actually safe / does it work given classloaders? Classes loaded by different classloaders are supposed to not pass `instanceof` checks against each other afaik, but I'm not sure where the `JsonConverter` comes from in this case. Basically, I'm not sure we'll actually enter this block when `JsonConverter` is loaded via `plugin.path`.

Have we tested outside of unit/integration tests that this works e2e? I think we might not hit an issue with this in practice anywhere just because connect-json ships on the same CLASSPATH as the connect-runtime, connect-api, etc, but if we ever factored out the pluggable stuff that is included w/ the framework to be loaded via separate classloaders, I think this would become a problem?

Reply:
Let’s test this to be sure, but a workaround for the `instanceof` test would be to check the fully qualified class names. If we go that route, it should go in a static helper method to keep the usage readable. That method would also be easier to test.

Reply:
@ewencp @rhauch 
Originally considered nixing the `instanceof` test in favor of a comparison with the fully-qualified class name, but it came in handy within the `PluginsTest` where `JsonConverter` could be subclassed and then its configuration properties could be examined to ensure correct behavior with things like `schemas.enable` defaulting to `false`.

Also considered adding a getter method to `JsonConverter` so that the value of `enableSchemas` can be examined for testing, but adding public API that people may end up using and relying on just so that unit testing could succeed seemed like a bad tradeoff.

If there's a better way to unit test the functionality we want to establish for defaulting `schemas.enable` to false, or if it's more worth it to abandon testing in favor of using a fully qualified class name over an `instanceof` check, I'd be happy to change it. Definitely not 100% satisfied with that line as it currently stands.